jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -350,17 +351,68 @@ class XGBoostGeneralSuite extends FunSuite with TmpFolderPerSuite with PerTest {
     val modelPath = getClass.getResource("/model/0.82/model").getPath
     val model = XGBoostClassificationModel.read.load(modelPath)
     val r = new Random(0)
-    val df = ss.createDataFrame(Seq.fill(100000)(1).map(i => (i, i))).
+    var df = ss.createDataFrame(Seq.fill(100000)(1).map(i => (i, i))).
       toDF("feature", "label").repartition(5)
+    // 0.82/model was trained with 251 features. and transform will throw exception
```

Comment:
based on my comments, this claim is invalid now, 

you should not use native  api to access model files produced by Spark directly

Reply:
Hi @CodingCat, I added some log info in native when loading 0.82/model from jvm, the model did store the feature size = 251.

Reply:
Just to put more context here.  The prediction data has only 1 feature, while model is trained with 251 features.